rpc/client/http/http.go

```diff
@@ -501,6 +501,34 @@ func (c *baseRPCClient) TxSearch(
 	return result, nil
 }
 
+func (c *baseRPCClient) TxSearchMatchEvents(
+	ctx context.Context,
+	query string,
+	prove bool,
+	page,
+	perPage *int,
+	orderBy string,
+	matchEvents bool,
+) (*ctypes.ResultTxSearch, error) {
+	if matchEvents {
+		query = query + " AND match.events = 1"
+	}
+	return c.TxSearch(ctx, query, prove, page, perPage, orderBy)
```

Comment:
I kept the call to `TxSearch` with the change in the query syntax to avoid adding a new method to the `client`. We could have a `TxSearchMatchEvents` in the Client interface and implement it as you suggested afterwards. But this again introduces a method into the interface. We could also have the method here not in the client interface but just in the implementations that exist. But to me that is somehow not clean.

Reply:
Implementation changed, this function does not exist anymore.